src/main/java/edu/kit/hci/soli/controller/LoginControllerAdvice.java

```diff
@@ -28,27 +29,14 @@ public CsrfToken getCsrfToken(HttpServletRequest request) {
     }
 
     @ModelAttribute("login")
-    public LoginStateModel getLoginStateModel(Principal principal, @AuthenticationPrincipal OidcUser oidcUser, HttpServletRequest request, @ModelAttribute("csrf") CsrfToken csrf) {
+    public LoginStateModel getLoginStateModel(Principal principal, @AuthenticationPrincipal User user, HttpServletRequest request, @ModelAttribute("csrf") CsrfToken csrf) {
+
         if (principal == null) {
             return new LoginStateModel("Visitor", LoginStateModel.Kind.VISITOR, csrf, null);
```

Comment:
Since the user is null in this case, passing it here should be fine (and might allow us to extend this to refer to some anonymous user later)